src/backend.rs

```diff
@@ -891,9 +952,16 @@ impl State {
         let mut size = 0;
         let mut channels = vec![];
         {
+            let db_path = self.db_path.clone();
             Self::with_channels_mut(selectors, &mut self.getter_by_id, |mut data| {
                 // This channel has changed, we may need to update watches.
                 if data.insert_tags(&tags) {
+                    if let Some(ref path) = db_path {
+                        let store = TagStorage::new(&path);
```

Comment:
So, when adding tags to several channels, we open the database for each channel? That looks a bit scary.